app/routers/types.py

```diff
@@ -0,0 +1,17 @@
+from typing import Generic, TypeVar
+
+from pydantic import BaseModel
+
+M = TypeVar("M", bound=BaseModel)
+
+
+class Pagination(BaseModel):
+    page: int
+    limit: int
+    total: int
+
+
+class ListResponse(BaseModel, Generic[M]):
```

Comment:
Minor, but the new syntax in python 3.12 can be:
```suggestion
class ListResponse[M: BaseModel](BaseModel):
```
and you don't even need to define the TypeVar above.
Ref: https://docs.python.org/3/whatsnew/3.12.html#pep-695-type-parameter-syntax

Reply:
neat